### PR TITLE
Add ESS lbm Module

### PR DESCRIPTION
**Pull request by micheltakken**

### CODE REVIEW

File: src/architecture/Module.h [hunk 1]
```diff
@@ -82,6 +87,39 @@ class Module {
      * @returns What type the channel has.
      */
     ModuleType getModuleType() const;
+
+    virtual void lbmInit(T dynViscosity, T density) {}
```
Comment (micheltakken) [inline src/architecture/Module.h:91]: From here, the functions are not relevant to the architecture, but manipulate solver parameters. These should go in a `simulation/CFDSolver.h` header file. Same goes for lbmModule and essLbmModule, which should be renamed and moved to the `simulation` folder.

File: src/architecture/ModuleOpening.h
Comment (micheltakken) [inline src/architecture/ModuleOpening.h:1]: Rename `ModuleOpening.h` ?

File: src/architecture/Module.h
Comment (micheltakken) [inline src/architecture/Module.h:1]: From line 91, the functions are not relevant to the architecture, but manipulate solver parameters. These should go in a `simulation/CFDSolver.h` header file. Same goes for lbmModule and essLbmModule, which should be renamed and moved to the `simulation` folder.

**Comment by micheltakken** [comment]
All hybrid examples must be adapted according to the new separated definition: Module vs Simulator

**Comment by micheltakken** [comment]
> All hybrid examples must be adapted according to the new separated definition: Module vs Simulator

**Comment by micheltakken** [comment]
Update python bindings and examples to new separated definition: Module vs Simulator.

**Review by MariaEmmerich — APPROVED** [review]

**Pull request merged**